app/service/data_svc.py

```diff
@@ -69,6 +69,13 @@ def __init__(self, dao, utility_svc):
             await self.dao.create('core_group_map', dict(group_id=identifier, agent_id=agent[0]['id']))
         return 'Saved %s host group' % name
 
+    async def delete_group(self, group_id):
+        group = await self.dao.get('core_group', dict(id=group_id))
+        await self.dao.delete('core_group', dict(id=group_id))
+        await self.dao.delete('core_group_map', dict(group_id=group_id))
+        await self.dao.delete('core_operation', dict(host_group=group_id))
```

Comment:
should we delete all artifacts from the operation as well? probably... thinking tables like results, facts, etc.